src/server/protocol.ts

```diff
@@ -511,6 +511,7 @@ namespace ts.server.protocol {
         endLocation: Location;
         category: string;
         code: number;
+        source?: string;

```

Comment:
is this change unrelated?

Reply:
Yes, I think @sheetalkamat might have a better idea, but this un-breaks an excess property error we had in the server layer.

Reply:
#15296 added this so diagnostics origin can be distinguished.